nesta/production/orms/orm_utils.py

```diff
@@ -6,35 +6,144 @@
 from sqlalchemy.engine.url import URL
 from sqlalchemy.orm import sessionmaker
 from sqlalchemy.sql.expression import and_
+from nesta.production.luigihacks.misctools import find_filepath_from_pathstub
+from nesta.production.luigihacks.misctools import get_config
+from elasticsearch import Elasticsearch
 
 import pymysql
 import os
-
 import json
-
 import logging
 import time
 
-def filter_out_duplicates(db_env, section, database, Base, _class, data, 
+def assert_correct_config(test, config, key):
+    """Assert that config key and 'index' value are consistent with the
+    running mode.    
+    
+    Args:
+        test (bool): Are we running in test mode?
+        config (dict): Elasticsearch config file data.
+        key (str): The name of the config key.
+    """
+    err_msg = ("In test mode='{test}', config key '{key}' "
+               "must end with '{suffix}'")
+    if test and not key.endswith("_dev"):
+        raise ValueError(err_msg.format(test=test, key=key, suffix='_dev'))
+    elif not test and not key.endswith("_prod"):
+        raise ValueError(err_msg.format(test=test, key=key, suffix='_prod'))
+    index = config['index']
+    if test and not index.endswith("_dev"):
+        raise ValueError(f"In test mode the index '{key}' "
+                         "must end with '_dev'")
+
+
+def setup_es(es_mode, test_mode, reindex_mode, dataset, aliases=None):
+    """Retrieve the ES connection, ES config and setup the index 
+    if required.
+    
+    Args:
+        es_mode (str): One of "prod" or "dev".
+        test_mode (bool): Running in test mode?
+        reindex_mode (bool): Running in reindex mode?
```

Comment:
I know this is minor, but could this be called something else so it's not confused with ES reindexing process? `recreate_index` / `drop_and_recreate`?

Reply:
good idea - changed in favour of `drop_and_recreate`